### PR TITLE
Support py3

### LINKED ISSUE #10: Support for python 3.x
https://github.com/striglia/pyramid_swagger would benefit from 3.x support.

Partial progress made here: https://github.com/analogue/bravado-core/tree/py3x_support

### PR DESCRIPTION
**Pull request by bpicolo**
This is ready once Julian/jsonschema#203 is in a release. Until then there's a bit of test flakiness for py3.

Fixes #10

### CODE REVIEW

File: bravado_core/docstring.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def create_operation_docstring(op):
     # TODO: add 'examples' if available
     # TODO: good idea to identify the default response?
     responses = op_spec.get('responses')
-    for http_status_code, response_spec in iter(sorted(responses.iteritems())):
+    for http_status_code, response_spec in iter(sorted(responses.items())):
```
Comment (prat0318) [inline bravado_core/docstring.py]: We can use `six.iteritems` here : https://pythonhosted.org/six/#six.iteritems

File: bravado_core/marshal.py [hunk 2]
```diff
@@ -111,7 +111,7 @@ def marshal_object(swagger_spec, object_spec, object_value):
             type(object_value), object_value))
 
     result = {}
-    for k, v in object_value.iteritems():
+    for k, v in object_value.items():
```
Comment (prat0318) [inline bravado_core/marshal.py]: Same as above.

File: bravado_core/operation.py [hunk 3]
```diff
@@ -154,7 +154,7 @@ def construct_params(self, request, op_kwargs):
             marshal_param(param, param_value, request)
 
         # Check required params and non-required params with a 'default' value
-        for remaining_param in current_params.itervalues():
+        for remaining_param in current_params.values():
```
Comment (prat0318) [inline bravado_core/operation.py]: `six.itervalues`

File: bravado_core/param.py [hunk 4]
```diff
@@ -1,9 +1,14 @@
 from functools import partial
 import logging
-import urllib
+try:
```
Comment (prat0318) [inline bravado_core/param.py]: `from six.moves.urllib.parse import quote`

File: bravado_core/docstring.py (hunk 1 above)
Comment (bpicolo) [inline bravado_core/docstring.py]: we can, but me and spatel talked and reckoned it's not worth it. Speed and memory difference at our scales more or less negligible

File: bravado_core/param.py (hunk 4 above)
Comment (bpicolo) [inline bravado_core/param.py]: Why use six when the builtin is easy to use? : P

**Comment by analogue** [comment]
@benp thanks for picking this up and making it happen. lgtm

**Comment by dnephin** [comment]
lgtm (once the jsonschema dep is ready and tests are stable)

**Comment by bpicolo** [comment]
Bumped it and added a test difference for py3 for a new test : ) Should be good now methinks

File: bravado_core/operation.py (hunk 3 above)
Comment (prat0318) [inline bravado_core/operation.py]: i think if we have changed all `iteritems` to be used from `six` package, we should also change the `values()` to be changed to the `six` package to keep the consistency.

**Comment by prat0318** [comment]
lgtm apart from the `values` concern.

**Comment by prat0318** [comment]
:+1:

**Pull request merged**